client/upe-onboarding-wizard/upe-preview-methods-selector/setup-complete-task.js

```diff
@@ -22,19 +20,27 @@ const SetupCompleteMessaging = () => {
 		return null;
 	}
 
-	const addedPaymentMethodsCount =
-		enabledMethodsCount - initialMethods.length;
+	// we need to check that the type of `enableUpePreviewPayload` is an object - it can also just be `true` or `undefined`
+	let addedPaymentMethodsCount = 0;
+	if (
+		typeof enableUpePreviewPayload === 'object' &&
+		enableUpePreviewPayload.initialMethods
+	) {
+		const { initialMethods } = enableUpePreviewPayload;
+		addedPaymentMethodsCount = enabledMethodsCount - initialMethods.length;
+	}
 
 	// can't just check for "0", some methods could have been disabled
 	if ( addedPaymentMethodsCount <= 0 ) {
 		return __( 'Setup complete!', 'woocommerce-gateway-stripe' );
 	}
 
+	// eslint-disable-next-line @wordpress/valid-sprintf
 	return sprintf(
-		/* translators: Number of payment methods */
+		/* translators: %s: the number of payment methods added */
 		_n(
-			'Setup complete! %d new payment method is now live on your store!',
-			'Setup complete! %d new payment methods are now live on your store!',
+			'Setup complete! One new payment method is now live on your store!',
+			'Setup complete! %s new payment methods are now live on your store!',
```

Comment:
As it only compares the number of methods, not the method ids, it does not consider deletion+ addition case. 

Suppose I had `card` and `giropay` before. Now if I remove `giropay` and add `eps`, `ideal` I have added 2 new methods but the message says `One` new payment method is now in my store.

Reply:
I mostly copied the functionality from https://github.com/Automattic/woocommerce-payments/blob/develop/client/additional-methods-setup/upe-preview-methods-selector/setup-complete-task.js#L22-L57 but you're right, I didn't think of that. I guess we'll have to make a more complex comparison.
Looking into it.

Reply:
Fixed in e8a358eb